unixfs/hamt/hamt.go

```diff
@@ -0,0 +1,466 @@
+package hamt
+
+import (
```

Comment:
To make it easier for people reading the code in the future it would be great to have a comment here at the top describing

- the exact type of HAMT that is implemented
- specific decisions made for this implementation